luigi_pipeline/lib/model/sv_mt_schema.py

```diff
@@ -196,17 +213,28 @@ class SeqrSVGenotypesSchema(SeqrGenotypesSchema):
 
     def _genotype_fields(self):
         is_called = hl.is_defined(self.mt.GT)
+        was_previously_called = hl.is_defined(self.mt.CONC_ST) & ~self.mt.CONC_ST.contains("EMPTY")
+        num_alt = self._num_alt(is_called)
+        prev_num_alt = PREVIOUS_GENOTYPE_N_ALT_ALLELES.get(hl.set(self.mt.CONC_ST), -1)
```

Comment:
I fixed the issues I was having by lifting this out of an `hl.if_else`.  I think it's cleaner anyways... though maybe a bit implicit that we allow ourselves to look up the `hl.set` missing value.

Reply:
Is there any expected case where `CONC_ST` does not match one of the defined keys in `PREVIOUS_GENOTYPE_N_ALT_ALLELES`? If not, I would rather not use a safe get with a default and instead have the pipeline run fail on an unexpected input

Reply:
👍 failing on unexpected is good by me!

Reply:
@hanars 

Got this working but it was painful.  Ideally we'd have:
```
prev_num_alt = hl.or_missing(PREVIOUS_GENOTYPE_N_ALT_ALLELES[hl.set(self.mt.CONC_ST)])
```
but that's gonna fail until hail releases the bug fix.  I put the missing value and `["EMPTY"]` in the lookup for now with a comment.  Are we ok with that for now (I'll make a ticket to fix pending hail release)?

Reply:
I see what you mean about this being ugly, would leaving the missing/empty values out of the dict and then doing something like so you only use the lookup expression when there is a valid value work?
```
prev_num_alt = PREVIOUS_GENOTYPE_N_ALT_ALLELES[hl.set(self.mt.CONC_ST)]
discordant_genotype = was_previously_called & (num_alt != prev_num_alt) & (prev_num_alt > 0)
novel_genotype = hl.or_else(~was_previously_called, (num_alt != prev_num_alt) & (prev_num_alt == 0))
```

Reply:
Hrm.... still having trouble.

I've tried:

```
was_previously_called = hl.is_defined(self.mt.CONC_ST) & ~self.mt.CONC_ST.contains("EMPTY")
prev_num_alt = PREVIOUS_GENOTYPE_N_ALT_ALLELES[hl.set(self.mt.CONC_ST)]
discordant_genotype = was_previously_called & (num_alt != prev_num_alt) & (prev_num_alt > 0)
novel_genotype = was_previously_called & (num_alt != prev_num_alt) & (prev_num_alt == 0)

// and
'prev_num_alt': hl.or_missing(discordant_genotype, prev_num_alt),
```

and the lookup is called.

Reply:
Still playing around though.

Reply:
Standard short circuiting _does_ seem to work in the way we'd expect:

```
In [17]: a = hl.dict({"1": 2})

In [18]: b = hl.dict({"2": 1})

In [19]: hl.eval(a.contains("2") & (b["3"] == 1))
Out[19]: False
```

Reply:
Aha

Here's more:

```
In [16]: a = hl.dict({"1": 2})

In [17]: b = hl.dict({"2": 1})

In [18]: c = a.contains("2")

In [19]: d = a.contains("1")

In [20]: e = b["3"]

In [21]: hl.eval(c & d & (e > 0)) // c is False so we never evaluate the KeyError in `e`
Out[21]: False

In [22]: hl.eval(c & d & (e > 0) & (b["3"] == a["1"])) // This is also fine.
Out[22]: False

In [23]: hl.eval(c & d & (e > 0) & (e == a["1"])) // This crashes since `e` seems to be evaluated.
```